### PR TITLE
add registered cluster api

### PR DESCRIPTION
**Pull request by leena-jawale**
Add registeredcluster api

### CODE REVIEW

File: api/singapore/v1alpha1/registeredcluster_types.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+}
+
+// RegisteredClusterStatus defines the observed state of RegisteredCluster
+type RegisteredClusterStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//ImportCommandRef is reference to configmap containing import command.
+	ImportCommandRef importCommandRef `json:"importCommandRef"`
```
Comment (robinbobbitt) [inline api/singapore/v1alpha1/registeredcluster_types.go:31]: In my head this was a `LocalObjectReference` versus just a string. Thoughts?

File: api/singapore/v1alpha1/registeredcluster_types.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+type RegisteredCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RegisteredClusterSpec   `json:"spec,omitempty"`
+	Status RegisteredClusterStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// RegisteredClusterList contains a list of ClusterRegistrar
```
Comment (robinbobbitt) [inline api/singapore/v1alpha1/registeredcluster_types.go:74]: list of RegisteredCluster

File: api/singapore/v1alpha1/registeredcluster_types.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+type RegisteredClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of ClusterRegistrar.
```
Comment (robinbobbitt) [inline api/singapore/v1alpha1/registeredcluster_types.go:80]: RegisteredCluster

File: api/singapore/v1alpha1/registeredcluster_types.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+// RegisteredClusterStatus defines the observed state of RegisteredCluster
+type RegisteredClusterStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//ImportCommandRef is reference to configmap containing import command.
+	ImportCommandRef importCommandRef `json:"importCommandRef"`
+
+	// Conditions contains the different condition statuses for this ClusterRegistrar.
```
Comment (robinbobbitt) [inline api/singapore/v1alpha1/registeredcluster_types.go:33]: RegisteredCluster

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
This looks great. Just one fix and some typos... Thanks!

File: api/singapore/v1alpha1/registeredcluster_types.go (hunk 1 above)
Comment (leena-jawale) [reply api/singapore/v1alpha1/registeredcluster_types.go:31]: yes we can use `LocalObjectReference`.

**Review by robinbobbitt — APPROVED** [review]
Awesome! Thank you @leena-jawale .

**Pull request merged**